source/module.c

```diff
@@ -516,6 +516,27 @@ PyObject *aws_py_memory_view_from_byte_buffer(struct aws_byte_buf *buf) {
     return PyMemoryView_FromMemory(mem_start, mem_size, PyBUF_WRITE);
 }
 
+PyObject *aws_py_weakref_get_ref(PyObject *object) {
+    PyObject *self = Py_None;
+#if PY_VERSION_HEX >= 0x030D0000               /* Check if Python version is 3.13 or higher */
+    if (PyWeakref_GetRef(object, &self) < 0) { /* strong reference */
+        return Py_None;
+    }
+#else
+    /* PyWeakref_GetObject is deprecated since python 3.13 */
+    self = PyWeakref_GetObject(object); /* borrowed reference */
```

Comment:
I vote we just have it behave like the new-function in ALL versions and return a New reference

Otherwise we live in a confusing world where it behaves 2 different ways, based on the version of python that built the wheel (not necessarily the version of python the user is running)